netplan_cli/cli/commands/status.py

```diff
@@ -88,18 +88,164 @@ def pretty_print(self, data: JSON, total: int, _console_width=None) -> None:
+        return []
+
+    def _get_missing_system_nameservers(self, ifname) -> list[str]:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('system_state', {}).get('missing_nameservers_addresses', [])
+        return []
+
+    def _get_missing_netplan_nameservers(self, ifname) -> list[str]:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('netplan_state', {}).get('missing_nameservers_addresses', [])
+        return []
+
+    def _get_missing_netplan_search(self, ifname) -> list[str]:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('netplan_state', {}).get('missing_nameservers_search', [])
+        return []
+
+    def _get_missing_system_search(self, ifname) -> list[str]:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('system_state', {}).get('missing_nameservers_search', [])
+        return []
+
+    def _get_missing_system_macaddress(self, ifname) -> str:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('system_state', {}).get('missing_macaddress')
+        return None
+
+    def _get_missing_netplan_routes(self, ifname) -> set[NetplanRoute]:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('netplan_state', {}).get('missing_routes', set())
+        return set()
+
+    def _get_missing_system_routes(self, ifname) -> set[NetplanRoute]:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('system_state', {}).get('missing_routes', set())
+        return set()
+
+    def _is_missing_dhcp4_address(self, ifname) -> bool:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('system_state', {}).get('missing_dhcp4_address', False)
+        return False
+
+    def _is_missing_dhcp6_address(self, ifname) -> bool:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('system_state', {}).get('missing_dhcp6_address', False)
+        return False
+
+    def _get_missing_system_bond_link(self, ifname) -> str:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('system_state', {}).get('missing_bond_link', '')
+        return ''
+
+    def _get_missing_netplan_bond_link(self, ifname) -> str:
+        if self.state_diff:
+            if diff := self.state_diff['interfaces'].get(ifname):
+                return diff.get('netplan_state', {}).get('missing_bond_link', '')
+        return ''
```

Comment:
nitpick (non-blocking): This (and surrounding) methods feel like lots of duplicated code, doing almost the same thing for `system_state` and `netplan_state`, I wonder if we could pass the relevant `diff.get('NETPLAN/SYSTEM_state')` data as an input variable, to use the same method for both cases?